nomad/csi_endpoint.go

```diff
@@ -140,23 +143,25 @@ func (v *CSIVolume) List(args *structs.CSIVolumeListRequest, reply *structs.CSIV
 				if raw == nil {
 					break
 				}
-
 				vol := raw.(*structs.CSIVolume)
-				vol, err := state.CSIVolumeDenormalizePlugins(ws, vol.Copy())
-				if err != nil {
-					return err
-				}
 
-				// Remove (possibly again) by PluginID to handle passing both NodeID and PluginID
+				// Remove (possibly again) by PluginID to handle passing both
+				// NodeID and PluginID
 				if args.PluginID != "" && args.PluginID != vol.PluginID {
 					continue
 				}
 
-				// Remove by Namespace, since CSIVolumesByNodeID hasn't used the Namespace yet
+				// Remove by Namespace, since CSIVolumesByNodeID hasn't used
+				// the Namespace yet
 				if vol.Namespace != ns {
 					continue
 				}
 
+				vol, err := snap.CSIVolumeDenormalizePlugins(ws, vol.Copy())
```

Comment:
what is the significance of copying the volume here, but not during `Get`?

Reply:
The `CSIVolumeByID` query in `Get` does a `CSIVolumeDenormalizePlugins` in the state store, so we already copy in the state store function. (I'm not wild about how that works currently because it's too easy to miss... I didn't want to get _too_ wild with the refactoring in this PR though.)